static/scenes/railroad/js/present.js

```diff
@@ -20,38 +26,46 @@ class Present {
                 }
               }
             this.model = obj;
-            this.scene.getScene().add(obj);
-            console.log("Present added");
+            this.scene.camera.add(obj);
+            obj.position.set(.125, -.125, -1);
         });
     }
 
     shoot(targetPosition) {
-        this.targetPosition = targetPosition;
-        this.startPosition = this.model.position;
+        this.scene.scene.attach(this.model);
+        this.targetPosition = targetPosition.clone();
+        this.startPosition = new THREE.Vector3();
+        this.model.getWorldPosition(this.startPosition);
+
+        var midpoint = this.startPosition.clone()
+            .multiplyScalar(1-midpointT)
+            .addScaledVector(this.targetPosition, midpointT)
+            .add(new THREE.Vector3(0, this.startPosition.distanceTo(this.targetPosition) * midpointOffset));
+
+        this.curve = new THREE.QuadraticBezierCurve3(this.startPosition.clone(), midpoint.clone(), this.targetPosition.clone());
 
-        var midpoint = new THREE.Vector3().add(this.targetPosition, this.startPosition).multiplyScalar(0.5);
-        midpoint.setY(midpoint.y + 3);
-        this.curve = new THREE.QuadraticBezierCurve3(this.startPosition, midpoint, this.targetPosition);
         this.inFlight = true;
     }
 
+    addDebugGeo() {
+        const points = this.curve.getPoints(50);
+        const geometry = new THREE.BufferGeometry().setFromPoints(points);
+        const material = new THREE.LineBasicMaterial({color: 0x00ff00});
+        const curveObject = new THREE.Line(geometry, material);
+        this.scene.scene.add(curveObject);
+    }
+
     update(seconds, deltaSeconds) {
         if (this.inFlight) {
             if (this.currentFlightTime > this.totalFlightTime) {
                 this.landed = true;
+                this.model.position.copy(this.targetPosition);
                 // call some kind of function to update score if good hit
             } else if (this.model) {
```

Comment:
Just noticed, after my PR this will always be true, so we can remove this if statement.

Reply:
Which if? this.inFlight?

Reply:
this.model will always be true after this: https://github.com/google/santa-tracker-web/pull/142